src/port1.0/portconfigure.tcl

```diff
@@ -515,17 +515,16 @@ proc portconfigure::configure_get_sdkroot {sdk_version} {
 
 # internal function to determine DEVELOPER_DIR according to Xcode dependency
 proc portconfigure::configure_get_developer_dir {} {
-    global use_xcode macosx_version
-    if {[vercmp $macosx_version 10.9] >= 0} {
```

Comment:
As @MarcusCalhoun-Lopez suggested, check for `/usr/lib/libxcselect.dylib` instead of macOS version. Presence of libxcselect indiciates the earliest version of macOS that places CLT in `/Library/Developer/CommandLineTools`. This needs to be tested.